File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        map<char,int>temp;
        for(char c:s)temp[c]++;
        map<int,vector<char>>temp2;
        for(auto lt:temp)
        {
            temp2[lt.second].push_back(lt.first);
        }
        string out="";
        for(auto lt:temp2){
            for(char c:lt.second){
                out+=string(lt.first,c);
            }
        }
        reverse(out.begin(),out.end());
        return out;
    }
};

File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        k=collections.defaultdict(int)
        for i in s:
            k[i]+=1
        out=''.join([i*k[i] for i in sorted(k.keys(),key=lambda x:k[x], reverse=True)])
        return out